src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,226 @@
+        }
+
+        public override string KeyExchangeAlgorithm => _impl.KeyExchangeAlgorithm;
+
+        public override int KeySize
+        {
+            get { return _impl.KeySize; }
+            set
+            {
+                // Perform the check here because LegalKeySizes are more restrictive here than _impl
+                if (!value.IsLegalSize(s_legalKeySizes))
+                    throw new CryptographicException(SR.Cryptography_InvalidKeySize);
+
+                _impl.KeySize = value;
+            }
+        }
+
+        public override KeySizes[] LegalKeySizes => s_legalKeySizes.CloneKeySizesArray();
+
+        // PersistKeyInCsp has no effect in Unix
+        public bool PersistKeyInCsp { get; set; }
+
+        public bool PublicOnly
+        {
+            get { return _publicOnly; }
+        }
+
+        public override string SignatureAlgorithm => _impl.SignatureAlgorithm;
+
+        public byte[] SignData(byte[] buffer)
+        {
+            return _impl.SignData(buffer, HashAlgorithmName.SHA1);
+        }
+
+        public byte[] SignData(byte[] buffer, int offset, int count)
+        {
+            return _impl.SignData(buffer, offset, count, HashAlgorithmName.SHA1);
+        }
+
+        public byte[] SignData(Stream inputStream)
+        {
+            return _impl.SignData(inputStream, HashAlgorithmName.SHA1);
+        }
+
+        public override string ToXmlString(bool includePrivateParameters) =>
+            _impl.ToXmlString(includePrivateParameters);
+
+        public override byte[] SignData(byte[] data, int offset, int count, HashAlgorithmName hashAlgorithm)
+        {
+            if (hashAlgorithm != HashAlgorithmName.SHA1)
+                throw new CryptographicException(SR.Cryptography_UnknownHashAlgorithm);
+
+            return _impl.SignData(data, offset, count, hashAlgorithm);
+        }
+
+        public override byte[] SignData(Stream data, HashAlgorithmName hashAlgorithm)
+        {
+            if (hashAlgorithm != HashAlgorithmName.SHA1)
+                throw new CryptographicException(SR.Cryptography_UnknownHashAlgorithm);
+
+            return _impl.SignData(data, hashAlgorithm);
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Security", "CA5351", Justification = "This is the implementation of DSACryptoServiceProvider")]
+        public byte[] SignHash(byte[] rgbHash, string str)
+        {
+            if (rgbHash.Length != SHA1_HASHSIZE)
+                throw new CryptographicException(string.Format(SR.Cryptography_InvalidHashSize, "SHA1", SHA1_HASHSIZE));
+            if (PublicOnly)
+                throw new CryptographicException(SR.Cryptography_CSP_NoPrivateKey);
+
+            // Only SHA1 allowed; the default value is SHA1
+            if (str != null && string.Compare(str, "SHA1", StringComparison.OrdinalIgnoreCase) != 0)
+                throw new CryptographicException(SR.Cryptography_UnknownHashAlgorithm);
+
+
```

Comment:
Ah, I lined up the throws and missed that the first if didn't match.  You missed the nullcheck, so this nullrefs on null rgbHash.